Domain/Do-Svyazi.User.Domain/Chats/Chat.cs

```diff
@@ -8,10 +8,72 @@ public abstract class Chat
     protected Chat() { }
 
     public Guid Id { get; protected init; } = Guid.NewGuid();
-    public string Name { get; init; }
-    public string Description { get; init; }
+    public string Name { get; protected set; }
+    public string Description { get; protected set; }
 
     // public long Tag { get; init; } ??
     public List<ChatUser> Users { get; init; } = new ();
     public List<Role> Roles { get; init; } = new ();
+
+    public virtual void ChangeNameChat(string name)
+    {
+        Name = name ?? throw new NullReferenceException("Name chat is null");
```

Comment:
`NullReferenceException` - это когда ты обращаешься к null _(например `name.Length` когда name - null)_, если мы работаем со строками, тут должен быть `ArgumentNullException` и проверка **строки** на то что она пустая, а не только на то что это `null`.

```suggestion
        if (string.IsNullOrWhiteSpace(name))
            throw new ArgumentNullException("Chat name to set is null");

        Name = name;
```

Ну и желательно чуть правильнее писать сообщение ошибки: Chat name is null || Chat name to set is null